Mathlib/Topology/Compactness/Lindelof.lean

```diff
@@ -0,0 +1,359 @@
+/-
+Copyright (c) 2023 Josha Dekker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Josha Dekker
+-/
+import Mathlib.Topology.Bases
+import Mathlib.Order.Filter.CountableInter
+import Mathlib.Topology.Compactness.Compact
+
+/-!
+# Lindelöf sets and Lindelöf spaces
+
+## Main definitions
+
+We define the following properties for sets in a topological space:
+
+* `IsLindelof`: a set such that each open cover has a countable subcover. This is defined in Mathlib
+  using filters.
+* `LindelofSpace`: typeclass stating that the whole space is a Lindëlof set.
+* `NonLindelofSpace`: a space that is not a Lindëlof space.
+
+## Main results
+
+* ToBeAdded
+-/
+open Set Filter Topology TopologicalSpace
+
+
+universe u v
+
+variable {X : Type u} {Y : Type v} {ι : Type*}
+
+variable [TopologicalSpace X] [TopologicalSpace Y] {s t : Set X}
+
+section Lindelof
+
+/-- A set `s` is Lindelöf if every open cover has a countable subcover. This is implemented in
+  Mathlib by showing that for every nontrivial filter `f` with the countable intersection
+  property that contains `s`, there exists `a ∈ s` such that every set of `f`
+  meets every neighborhood of `a`. The equivalence of these two still needs to be proven in Mathlib
+  (Work in progress). -/
+def IsLindelof (s : Set X) :=
+  ∀ ⦃f⦄ [NeBot f] [CountableInterFilter f], f ≤ 𝓟 s → ∃ x ∈ s, ClusterPt x f
+
+/-- Type class for Lindelöf spaces.  -/
+class LindelofSpace (X : Type*) [TopologicalSpace X] : Prop where
+  /-- In a Lindelöf space, `Set.univ` is a Lindelöf set. -/
+  isLindelof_univ : IsLindelof (univ : Set X)
+
```

Comment:
This would be a good place to put two lemmas
- "X is a Lindelöf space iff univ is Lindelöf
- -if X is Lindelöf, univ is

These are mathematically obvious from your definition; such lemmas will be useful for working with Lindelöf spaces in mathlib.

Reply:
Added!